source/elements/oneMKL/source/domains/spblas/data_types/create_coo_matrix.rst

```diff
@@ -0,0 +1,241 @@
+incremented which extends the lifetime of the underlying buffer until the
+``smhandle`` is destroyed with ``destroy_sparse_matrix`` or the data is reset with
+``set_coo_matrix_data``.
+
+In the case of USM, the object does not take ownership of the data.
+
+The ``oneapi::mkl::sparse::create_coo_matrix`` function defined below takes in
+the number of non-zero elements in the sparse matrix as an argument. However, in
+certain math operations where the output is a sparse matrix, e.g., sparse matrix
+addition (sparse matrix + sparse matrix = sparse matrix), and multiplication of
+two sparse matrices, the number of non-zero elements in the output sparse matrix
+is not known in advance and must be calculated as part of the operation/API.
+Such APIs are currently not a part of the current oneMKL Specification, but will
+be added in the future. Therefore, it is important for the ``create_coo_matrix``
+API to be able to handle multiple calls to build the sparse matrix as function
+arguments become known and arrays are allocated with the correct sizes. In
+particular, the ``create_coo_matrix`` API must handle the case where the number
+of non-zero elements in the matrix are unknown and the matrix arrays are
+``nullptr`` or zero-sized ``sycl::buffer`` objects as applicable. This behavior
+is currently left to be implementation-defined, but may be clarified in the
+oneMKL Specification in the future.
+
+See :ref:`onemkl_sparse_matrix_handle`.
+
+.. _onemkl_sparse_create_coo_matrix_buffer:
+
+create_coo_matrix (Buffer version)
+----------------------------------
+
+.. rubric:: Syntax
+
+.. code-block:: cpp
+
+   namespace oneapi::mkl::sparse {
+
+       template <typename dataType, typename indexType>
+       void create_coo_matrix (sycl::queue                          &queue,
+                               oneapi::mkl::sparse::matrix_handle_t *p_smhandle,
+                               std::int64_t                         num_rows,
+                               std::int64_t                         num_cols,
+                               std::int64_t                         nnz,
+                               index_base                           index,
+                               sycl::buffer<indexType, 1>           &row_ind,
+                               sycl::buffer<indexType, 1>           &col_ind,
+                               sycl::buffer<dataType, 1>            &val);
+
+   }
+
+.. container:: section
+
+   .. rubric:: Template parameters
+
+   dataType
+      See :ref:`supported template types<onemkl_sparse_supported_types>`.
+
+   indexType
+      See :ref:`supported template types<onemkl_sparse_supported_types>`.
+
+.. container:: section
+
+   .. rubric:: Input parameters
+
+   queue
+      The SYCL command queue which will be used for SYCL kernels execution.
+
+   p_smhandle
+      The address of the ``p_smhandle`` object to be initialized. Must only be
+      called on an uninitialized ``matrix_handle_t`` object.
```

Comment:
> Must only be called on an uninitialized ``matrix_handle_t`` object.

This conflicts with what we have in the text description above saying that for multi-stage sparse-matrix-generating APIs, this API needs to be able to handle partially built matrix handles. I suppose this highlighted statement is still correct, so maybe we should clarify in the text description about partial handles above that a different API (is it set_coo_matrix_data()?) would be used to update and "complete" partially built matrix handles?

Reply:
I didn't see this as contradicting because I read this as the``matrix_handle_t`` object is uninitialized from a C++ point of view.

I understand it can be confusing. I think we should remove:
> Therefore, it is important for the ``create_coo_matrix``
API to be able to handle multiple calls to build the sparse matrix as function
arguments become known and arrays are allocated with the correct sizes. In
particular, the ``create_coo_matrix`` API must handle the case where the number
of non-zero elements in the matrix are unknown and the matrix arrays are
``nullptr`` or zero-sized ``sycl::buffer`` objects as applicable.

As you suggest `set_coo_matrix_data` would be more suitable in my opinion. I'd rather keep it vague for now and think about supporting this in the future.

Reply:
I've removed the restriction for `create_*` functions to be called multiple times in https://github.com/oneapi-src/oneAPI-spec/pull/522/commits/0f7c767fb6e936a5d0edbed5034dfaf779a9b244